File: 1mLeerCode/238. Product of Array Except Self.js
function productExceptSelf(nums) {
  const output = new Array(nums.length);
  let left = 1;
  for (let i = 0; i < nums.length; i++) {
    output[i] = left;
    left *= nums[i];
  }
  let right = 1;
  for (let j = nums.length - 1; j >= 0; j--) {
    output[j] *= right;
    right *= nums[j];
  }

  return output;
}

// Example usage:
const nums = [2, 2, 3];
console.log(productExceptSelf(nums)); // Output: [24, 12, 8, 6]

File: 1mLeerCode/238. Product of Array Except Self.py

def productExceptSelf(nums):
    length = len(nums)
    output = [1] * length
  
    left = 1
    for i in range(length):
      output[i] = left
      left *= nums[i]
    right = 1
    for j in reversed(range(length )):
      output[j] *= right
      right *= nums[j]
    return output 

nums = [1, 2, 3, 4]
print(productExceptSelf(nums))

